### ISSUE TITLE
"There was an error connecting to SteelSeries Engine 3"... but it works

### ISSUE THREAD

**Issue by vinnydiehl** [JayJay1989/GameSenseSDK#9]
Running SteelSeries Engine v3.17.8 on my SteelSeries X MSI laptop keyboard. Minecraft version 1.15.2 on Forge 31.1.63. Using [this mod version](https://www.curseforge.com/minecraft/mc-mods/steelseries-gamesense/files/2866202) from the CurseForge page, labeled 1.15.2. The tool functionality is slow to update (though it shows updates in the console every couple of seconds) and the default hunger bar flashes before it's even halfway empty, but I simply disabled the tool and set a custom hunger bar to flash at 25% and I'm happy. Will create a separate issue for these if you desire to keep tickets separate.

The problem is, every 30 seconds or so the error message `There was an error connecting to SteelSeries Engine 3` flashes in the chat, even though there is no lapse in functionality of the keyboard. This makes the mod basically unusable until this is fixed- I can't just hide the chat, I use lots of commands. Please at least allow the error to be disabled.

**To Reproduce**

1. Start up Minecraft and begin to play; keyboard lights function as expected
2. Receive error message frequently, despite the mod working perfectly

**Expected Behavior**

No error message.

**Log**

There isn't much information about the error in the console output. Just item updates every second, punctuated by the occasional log that the error has been printed to the chat.

```
[m[32m[05:25:49] [Render thread/INFO] [STDOUT/]: [com.sse3.gamesense.internal.EventReceiver:onLivingUpdate:108]: AxeItem
[m[32m[05:25:50] [Render thread/INFO] [STDOUT/]: [com.sse3.gamesense.internal.EventReceiver:onLivingUpdate:108]: AxeItem
[m[32m[05:25:51] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[05:25:51] [Render thread/INFO] [STDOUT/]: [com.sse3.gamesense.internal.EventReceiver:onLivingUpdate:108]: AxeItem
[m[32m[05:25:52] [Render thread/INFO] [STDOUT/]: [com.sse3.gamesense.internal.EventReceiver:onLivingUpdate:108]: AxeItem
```

**Minecraft Enviroment**
 - OS: Windows 10 Home Update 1809
 - Ram allocated: 20G
 - Minecraft & Forge version: 1.15.2 - 31.1.63
 - Mod Version: 1.15.2

**Comment by JayJay1989** [comment]
thank you for reporting this. I will look into this

**Comment by vinnydiehl** [comment]
Thank you for your fast response. For now I just removed the error message from the code and compiled it and it's working great! However this is definitely a hack and I'm unfamiliar with the inner workings of GameSense, so I'm curious what you come up with. If there are any other logs that I can access and provide please let me know.

It appears that the underlying cause may be a brief lapse of connection on SteelSeries's end? Since SendGameEvent() is being run so often, if there is even a moment where SteelSeries Engine is not responding, it sends a chat error. Recommend implementing a timeout that only sends the error if the Engine hasn't been responding for some time, or removing the error altogether as these ConnectTimeoutExceptions don't appear to be catastrophic.

I only looked at a small portion of the code so please let me know if there's anything I'm missing.

**Comment by JayJay1989** [comment]
i had the same error on the fabric mod. The events are sent by the server and client, i think.
I'll debug it and fix it in the next 3-4 hours.

**Comment by JayJay1989** [comment]
@vinnydiehl can you try the attached version?
[gamesense-1.15.2-1.13.22.jar.zip](https://github.com/JayJay1989/GameSenseSDK/files/4661532/gamesense-1.15.2-1.13.22.jar.zip)

**Comment by vinnydiehl** [comment]
The errors are still there but their behavior has changed. Instead of one every 30 seconds, I got 6 within a few seconds of loading a world, then 2 more about 10 minutes later.

```
[m[32m[16:19:21] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:19:21] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:19:36] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:19:38] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:19:46] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:19:46] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:27:11] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
[m[32m[16:27:28] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] There was an error connecting to SteelSeries Engine 3
```

I restarted and tried again, this time I got 4 errors within the first minute. The playerChatErrors config works as expected.

**Comment by JayJay1989** [comment]
@vinnydiehl its weird, i dont get any errors over here or my test environment . Are you sure u u se the latest version of 'Steelseries Gamesense' or have it at least installed and minecraft gamesense enabled

**Comment by vinnydiehl** [comment]
Yep, my SteelSeries Engine is the latest version (3.17.8), and my keyboard drivers are up to date (PS/2 v2.1.3.0). Minecraft GameSense is enabled under Apps. Forge is up to date (31.2.0).
